main.c:70:34: error: format '%d' expects argument of type 'int', but argument 2 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]    70 |     printf("State closed, heap: %d\n", xPortGetFreeHeapSize());

main.c:70:34: error: format '%d' expects argument of type 'int', but argument 2 has type 'uint32\_t' {aka 'long unsigned int'} \[-Werror=format=\]

70 | printf("State closed, heap: %d\\n", xPortGetFreeHeapSize());

--- a/examples/alloc/main/main.c
+++ b/examples/alloc/main/main.c
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <inttypes.h>
 #include <freertos/FreeRTOS.h>
 #include <freertos/task.h>
 #include <string.h>
@@ -73,7 +74,7 @@ void test(void *arg)
 
     lua_close(L);
 
-    printf("State closed, heap: %d\n", xPortGetFreeHeapSize());
+    printf("State closed, heap: %" PRIu32 "\n", xPortGetFreeHeapSize());
     while (1)
     {
         printf(".");

--- a/examples/custom_lib/main/main.c
+++ b/examples/custom_lib/main/main.c
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <inttypes.h>
 #include <freertos/FreeRTOS.h>
 #include <freertos/task.h>
 #include <string.h>
@@ -67,7 +68,7 @@ void test(void *arg)
 
     lua_close(L);
 
-    printf("State closed, heap: %d\n", xPortGetFreeHeapSize());
+    printf("State closed, heap: %" PRIu32 "\n", xPortGetFreeHeapSize());
     while (1)
     {
         printf(".");

--- a/examples/simple/main/main.c
+++ b/examples/simple/main/main.c
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <inttypes.h>
 #include <freertos/FreeRTOS.h>
 #include <freertos/task.h>
 #include <string.h>
@@ -27,7 +28,7 @@ static int report(lua_State *L, int status)
 void test(void *arg)
 {
     printf("\n\n%s\n\n", prg);
-    printf("Start, heap: %d\n", xPortGetFreeHeapSize());
+    printf("Start, heap: %" PRIu32 "\n", xPortGetFreeHeapSize());
 
     lua_State *L = luaL_newstate();
     if (!L)
@@ -36,11 +37,11 @@ void test(void *arg)
         while (1) { vTaskDelay(1000); }
     }
 
-    printf("State ready, heap: %d\n", xPortGetFreeHeapSize());
+    printf("State ready, heap: %" PRIu32 "\n", xPortGetFreeHeapSize());
 
     luaL_openlibs(L);
 
-    printf("Libs ready, heap: %d\n", xPortGetFreeHeapSize());
+    printf("Libs ready, heap: %" PRIu32 "\n", xPortGetFreeHeapSize());
 
     int r = luaL_loadstring(L, prg);
     if (r)
@@ -49,7 +50,7 @@ void test(void *arg)
         while (1) { vTaskDelay(1000); }
     }
 
-    printf("Prg loaded, heap: %d\n", xPortGetFreeHeapSize());
+    printf("Prg loaded, heap: %" PRIu32 "\n", xPortGetFreeHeapSize());
 
     printf("-------------------------------------------\n");
     r = lua_pcall(L, 0, 0, 0);
@@ -57,11 +58,11 @@ void test(void *arg)
         report(L, r);
     printf("-------------------------------------------\n");
 
-    printf("Prg done, heap: %d\n", xPortGetFreeHeapSize());
+    printf("Prg done, heap: %" PRIu32 "\n", xPortGetFreeHeapSize());
 
     lua_close(L);
 
-    printf("State closed, heap: %d\n", xPortGetFreeHeapSize());
+    printf("State closed, heap: %" PRIu32 "\n", xPortGetFreeHeapSize());
     while (1)
     {
         printf(".");

--- a/examples/vfs/main/main.c
+++ b/examples/vfs/main/main.c
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <inttypes.h>
 #include <freertos/FreeRTOS.h>
 #include <freertos/task.h>
 #include <string.h>
@@ -78,7 +79,7 @@ void test(void *arg)
     report(L, r);
     lua_close(L);
 
-    printf("State closed, heap: %d\n", xPortGetFreeHeapSize());
+    printf("State closed, heap: %" PRIu32 "\n", xPortGetFreeHeapSize());
 
     while (1)
     {

